src/js/components/sharedComponents/moreOptionsTabs/MoreOptionsTabs.jsx

```diff
@@ -0,0 +1,209 @@
+/**
+ * MoreOptionsTabs.jsx
+ * Created by James Lee 6/18/20
+ */
+
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { throttle } from 'lodash';
+import { Picker } from 'data-transparency-ui';
+import { formatNumber } from 'accounting';
+import ResultsTableTabs from '../../search/table/ResultsTableTabs';
+
+const propTypes = {
+    tabs: PropTypes.array.isRequired,
+    tabCounts: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
+    classes: PropTypes.string,
+    pickerLabel: PropTypes.string,
+    changeActiveTab: PropTypes.func
+};
+
+const MoreOptionsTabs = (props) => {
+    const [activeTab, setActiveTab] = useState(props.tabs[0].internal);
+    const [pickerActiveTab, setPickerActiveTab] = useState('');
+    const [tabClass, setTabClass] = useState('');
+    const [showMoreOptions, setShowMoreOptions] = useState(false);
+    const [tabsContainerWidth, setTabsContainerWidth] = useState(null);
+    const [pickerOptions, setPickerOptions] = useState([]);
+    const [tabTypes, setTabTypes] = useState(props.tabs);
+    const [indexesToDelete, setIndexesToDelete] = useState([]);
+    const tabs = useRef(null);
+
+    const switchTab = (tab) => {
+        pickerOptions.forEach((option) => {
+            if (option.value === tab) {
+                setPickerActiveTab(tab);
+            }
+        });
+        setActiveTab(tab);
+
+        // update subtiitle based on tab selected
+        if (props.changeActiveTab) {
+            props.changeActiveTab(tab);
+        }
+    };
+
+    const filteredSelectedOption = pickerOptions.filter((option) => option.value === activeTab);
+    const filteredSelectedPickerOption = pickerOptions.filter((option) => option.value === pickerActiveTab);
+
+    const selectOption = (name, value) => (
+        <div
+            role="listbox"
+            onChange={switchTab}
+            onKeyDown={switchTab}
+            title={`Show ${name}`}
+            aria-label={`Show ${name} - ${props.tabCounts[value]} ${props.tabCounts[value] === 1 ? 'result' : 'results'}`}
+            tabIndex={0}>
+            <div className="tab-content">
+                <div className="tab-label">
+                    {name}
+                </div>
+                <div className={`count-badge ${value === activeTab ? 'active' : ''}`}>
+                    {formatNumber(props.tabCounts[value])}
+                </div>
+            </div>
+        </div>
+    );
+
+    const getIndexesToDelete = (allTabs, containerWidth, moreOptionsWidth) => {
```

Comment:
Wondering if some of these functions could be moved to a helper that could be unit tested outside the scope of this component